plan/plan.go

```diff
@@ -146,6 +146,14 @@ func (p *Plan) Calculate() *Plan {
 
 	for _, topRow := range t.rows {
 		for _, row := range topRow {
+			// We ignore defined Endpoints with empty targets, as there seems to be
+			// intention for the DNS entry, but it is not known where to point them.
+			// This is the case when the status is missing from ingresses because a
+			// failure in the ingress controller.
+			if endpoint.EndpointsHaveEmptyTargets(row.candidates) {
+				continue
+			}
```

Comment:
Some general thoughts:

1. I'm concerned about the desire to work-around specific ingress controller failures in external-dns. I think there are other use-cases / value for something like this, but let's make sure we understand those use-cases that don't involve a bug in a third-party.
2. I highly recommend this be a feature that is properly gated by a configuration mechanism. Either a startup flag (like the existing upsert-only policy) or an annotation (like the existing `external-dns.alpha.kubernetes.io/target` override). Something like `ignore-empty-targets`.

Reply:
Added `ignore-empty-targets` as a parameter in order to use this feature in 6f675a02daec011ef0f72cdb31891114a5e49047.

Reply:
> 1. I'm concerned about the desire to work-around specific ingress controller failures in external-dns. I think there are other use-cases / value for something like this, but let's make sure we understand those use-cases that don't involve a bug in a third-party.

I have been searching but I am not sure there is a different use-case than resilience. What is the use-case for `upsert-only`? My suspicion is that the main use-case for `upsert-only` is a work-around for this bug that was and still is in one of the two nginx ingress controllers. That begs the question: Why would you support that, but not this one? We can live without this change, but it would make our cluster more resilient against other problems that might mess with the targets.

Why is external-dns a big deal? Because with DNS TTLs it can take some time to recover opposed to a component that quickly restarts.

As we know when we want to fail-over to another cluster (how often does that happen?) we could simply disable this resilience switch in such a situation.

If resilience against a real problem not a good enough justification then we will stop working on this. It seems to have happened often enough to implement `upsert-only` 

[kubernetes-nginx ](https://github.com/kubernetes/ingress-nginx) is still broken, as we do not use it but checked whether is was an option for a fix.

[The original PR](https://github.com/kubernetes-sigs/external-dns/pull/127#issue-219336677) of the `upsert-only` policy imagined a similar feature: "I can think of a policy that would reject an unusually large list of deletions that could indicate an error in another part of the system."

In a highly available distributed system should we not abandon this view: "If another component has a problem it is not our cup of tea"?